src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs

```diff
@@ -0,0 +1,373 @@
+
+        // Write scope tag and length
+        ProtobufSerializerHelper.WriteTagAndLengthPrefix(buffer, ref scopeMetricsIndex, currentPosition - scopeMetricsIndex - LengthAndTagSize, FieldNumberConstants.ResourceMetrics_scope_metrics, WireFormat.WireType.LengthDelimited);
+
+        // Write resource metric tag and length
+        ProtobufSerializerHelper.WriteTagAndLengthPrefix(buffer, ref resourceMetricIndex, currentPosition - resourceMetricIndex - LengthAndTagSize, FieldNumberConstants.ResourceMetrics_resource, WireFormat.WireType.LengthDelimited);
+    }
+
+    private void SendMetricPoint(byte[] buffer, ref int currentPosition)
+    {
+        // TODO: Extend this for user_events.
+        this.MetricDataTransport.SendOtlpProtobufEvent(buffer, currentPosition);
+    }
+
+    internal static void WriteInstrumentDetails(byte[] buffer, ref int currentPosition, Metric metric)
+    {
+        // Write metric name
+        ProtobufSerializerHelper.WriteStringTag(buffer, ref currentPosition, metric.Name, FieldNumberConstants.Metric_name);
+
+        // Write metric description
+        if (metric.Description != null)
+        {
+            ProtobufSerializerHelper.WriteStringTag(buffer, ref currentPosition, metric.Description, FieldNumberConstants.Metric_description);
+        }
+
+        // Write metric unit
+        if (metric.Unit != null)
+        {
+            ProtobufSerializerHelper.WriteStringTag(buffer, ref currentPosition, metric.Unit, FieldNumberConstants.Metric_unit);
+        }
+    }
+
+    internal static void SerializeInstrumentationScope(byte[] buffer, ref int currentPosition, string name, IEnumerable<KeyValuePair<string, object>> meterTags)
+    {
+        int previousPosition = currentPosition;
+
+        currentPosition += LengthAndTagSize;
+
+        // Write name
+        ProtobufSerializerHelper.WriteStringTag(buffer, ref currentPosition, name, 1);
+
+        SerializeTags(buffer, ref currentPosition, meterTags, FieldNumberConstants.InstrumentationScope_attributes);
+
+        // Write instrumentation Scope Tag
+        ProtobufSerializerHelper.WriteTagAndLengthPrefix(buffer, ref previousPosition, currentPosition - previousPosition - LengthAndTagSize, FieldNumberConstants.ScopeMetrics_scope, WireFormat.WireType.LengthDelimited);
+    }
+
+    private static void SerializeTags(byte[] buffer, ref int currentPosition, IEnumerable<KeyValuePair<string, object>> attributes, int fieldNumber)
+    {
+        if (attributes != null)
+        {
+            foreach (var tag in attributes)
+            {
+                if (tag.Value != null)
+                {
+                    SerializeTag(buffer, ref currentPosition, tag.Key, tag.Value, fieldNumber);
+                }
+            }
+        }
+    }
+
+    private static void SerializeResource(byte[] buffer, ref int currentPosition, Resource resource)
+    {
+        if (resource != null && resource.Attributes != null)
```

Comment:
We want to serialize Resources only if it has some attributes otherwise we would waste some space to serialize zero as the length of the resources.
```suggestion
        if (resource != Resource.Empty)
```

Reply:
We don't need the `null` check here. The public API [documentation](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/ProviderExtensions.cs#L21) states that it would return `Resource.Empty` when no resource is found.